cdap-ui/app/directives/dag-plus/my-dag.js

```diff
@@ -39,7 +39,7 @@ commonModule.directive('myDagPlus', function() {
     },
     link: function(scope, element) {
       scope.element = element;
-      scope.getGraphMargins = function (plugins) {
+      scope.getGraphMargins = function (plugins, moveErrorsAlerts = false) {
```

Comment:
What would be the scenario when, even though we have errors and alerts, we wouldn't want to move them?

Reply:
If we set this to true by default, then every call to `fitToScreen()` will move the error and alert nodes a little bit. It's the `cleanUpGraph()` (align) algorithm that needs to move these nodes, and inside this function we do call `getGraphMargins` with `moveErrorsAlerts` set to true.

Reply:
Hhmm it might time we revisit the alignment algorithm. It might not hold correct anymore as we have seen produce unpredictable layout for the same dag. The issue in the algorithm is forcing us to make this hack. IMO we should fix the algorithm rather than workarounds/hacks.